vector/src/main/java/im/vector/adapters/MediasPreviewAdapter.java

```diff
@@ -0,0 +1,149 @@
+        ImageView imagePreview;
+
+        ImageItemViewHolder(View itemView) {
+            super(itemView);
+            imagePreview = (ImageView) itemView.findViewById(R.id.image_preview);
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        // Return the amount of items in the list.
+        return mImagePreviewList.size();
+    }
+
+    @Override
+    public ImageItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        // Create a new CardItem when needed.
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.image_preview, parent, false);
+        ImageItemViewHolder imageItemViewHolder = new ImageItemViewHolder(view);
+        return imageItemViewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(ImageItemViewHolder holder, int position) {
+        final Context context = holder.imagePreview.getContext();
+        String mimeType = mImagePreviewList.get(position).getMimeType(context);
+        Uri uri = mImagePreviewList.get(position).getUri();
+
+        if (mimeType != null) {
+            if (mimeType.startsWith("image")) {
+                holder.imagePreview.setImageURI(uri);
+            } else if (mimeType.startsWith("video")) {
+                MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever();
+                mediaMetadataRetriever.setDataSource(context, uri);
+                Bitmap bitmap = mediaMetadataRetriever.getFrameAtTime();
+                holder.imagePreview.setImageBitmap(bitmap);
+            } else {
+                holder.imagePreview.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.filetype_attachment));
+                holder.imagePreview.setScaleType(ImageView.ScaleType.FIT_CENTER);
+            }
+        }
+
+        final RoomMediaMessage mediaMessage = mImagePreviewList.get(position);
+
+        // Change to the image that is clicked on.
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ViewGroup viewGroup = (ViewGroup) view.getParent().getParent();
+                ImageView imagePreview = (ImageView) viewGroup.findViewById(R.id.image_previewer);
+                WebView webPreview = (WebView) viewGroup.findViewById(R.id.web_previewer);
+                VideoView videoPreview = (VideoView) viewGroup.findViewById(R.id.video_previewer);
+                ImageView filePreview = (ImageView) viewGroup.findViewById(R.id.file_previewer);
+                TextView fileNameView = (TextView) viewGroup.findViewById(R.id.file_name);
+
+                setPreview(webPreview, imagePreview, videoPreview, filePreview, fileNameView, mediaMessage, context);
+            }
+        });
+    }
+
+    public void setPreview(WebView webPreview, ImageView imagePreview, final VideoView videoPreview, ImageView filePreview,
```

Comment:
Please use on line per parameter for code clarity, and move this code to the Activity
Also please implement a filter in the Activity, to avoid blink effect if I click several time on the same item